lib/dao.js

```diff
@@ -2407,6 +2574,197 @@ DataAccessObject.prototype.unsetAttribute = function unsetAttribute(name, nullif
 };
 
 /**
+ * Replace set of attributes.
+ * Performs validation before replacing.
+ *
+ * @trigger `validation`, `save` and `update` hooks
+ * @param {Object} data Data to replace
+ * @param {Object} [options] Options for replace
+ * @param {Function} cb Callback function called with (err, instance)
+ */
+DataAccessObject.prototype.replaceAttributes = function(data, options, cb) {
+  var Model = this.constructor;
+  var id = getIdValue(this.constructor, this);
+  return Model.replaceById(id, data, options, cb);
+};
+
+DataAccessObject.replaceById = function(id, data, options, cb) {
+  var connectionPromise = stillConnecting(this.getDataSource(), this, arguments);
+  if (connectionPromise) {
+    return connectionPromise;
+  }
+  var PKName = idName(this);
+  data[PKName] = id;
+  var Model = this;
+  var inst = new Model(data);
+  var enforced = {};
+  this.applyProperties(enforced, inst);
+  inst.setAttributes(enforced);
+  Model = this.lookupModel(data); // data-specific
+  if (Model !== inst.constructor) inst = new Model(data);
+  var strict = inst.__strict;
+  // delete data[PKName];
```

Comment:
In terms of expected behavior still not sure whether I need to delete `id` after creating instance or not; please note I'm using this `data` for `data` in `persist` hook which you asked me in [this comment](https://github.com/strongloop/loopback-datasource-juggler/pull/788#discussion-diff-47087458)

So now that I'm not deleting `id` from `data`in `data` from `persist` I expect `id` as well; please see [here](https://github.com/strongloop/loopback-datasource-juggler/pull/788/files#diff-145ab49855b38984ba190919e6e7397dR1412)